### PR TITLE
Duplicate assistant using /new

### PR DESCRIPTION
**Pull request by lasryaric**
Remake of the duplicate assistant for Ahuna. Original bad PR is here:
https://github.com/dust-tt/dust/pull/2917

### CODE REVIEW

File: front/components/assistant_builder/AssistantBuilder.tsx [hunk 1]
```diff
@@ -259,17 +261,41 @@ export default function AssistantBuilder({
...
-    scope: initialBuilderState?.scope ?? defaultScope,
-    generationSettings: {
-      ...DEFAULT_ASSISTANT_STATE.generationSettings,
-      modelSettings:
-        plan.code === FREE_TEST_PLAN_CODE
-          ? GPT_3_5_TURBO_MODEL_CONFIG
-          : GPT_4_TURBO_MODEL_CONFIG,
-    },
-  });
+  const [builderState, setBuilderState] = useState<AssistantBuilderState>(
```
Comment (lasryaric) [inline front/components/assistant_builder/AssistantBuilder.tsx:264]: Context: we want to inject the props `initialBuilderState` only once, after that the state is fully controller by the component, and not by the props.

Comment (lasryaric) [reply front/components/assistant_builder/AssistantBuilder.tsx:264]: Using useEffect() to control that was leading to a race condition between the setting of the initial state and the setting of a random avatar.

File: front/pages/w/[wId]/builder/assistants/new.tsx [hunk 2]
```diff
@@ -70,9 +126,52 @@ export default function CreateAssistant({
...
+  dustAppConfiguration,
+  databaseQueryConfiguration,
+  agentConfiguration,
   flow,
 }: InferGetServerSidePropsType<typeof getServerSideProps>) {
+  let actionMode: AssistantBuilderInitialState["actionMode"] = "GENERIC";
+
+  let timeFrame: AssistantBuilderInitialState["timeFrame"] = null;
+
+  if (agentConfiguration) {
```
Comment (lasryaric) [inline front/pages/w/[wId]/builder/assistants/new.tsx:140]: Same logic as in the edit file, but guarded by the existence of an `agentConfiguration`

File: front/pages/w/[wId]/builder/assistants/[aId]/index.tsx [hunk 3]
```diff
@@ -185,12 +132,91 @@ export const getServerSideProps: GetServerSideProps<{
...
+    !context.params?.aId
+  ) {
+    return {
+      notFound: true,
+    };
+  }
+
+  const allDataSources = await getDataSources(auth);
+
+  const dataSourceByName = allDataSources.reduce(
```
Comment (spolu) [inline front/pages/w/[wId]/builder/assistants/[aId]/index.tsx:182]: Why not move that in buildInitialState?

File: front/pages/w/[wId]/builder/assistants/new.tsx [hunk 4]
```diff
@@ -43,21 +63,57 @@ export const getServerSideProps: GetServerSideProps<{
   const allDataSources = await getDataSources(auth);
   const allDustApps = await getApps(auth);
 
+  const dataSourceByName = allDataSources.reduce(
```
Comment (spolu) [inline front/pages/w/[wId]/builder/assistants/new.tsx:66]: Since it's copied here?

File: front/pages/w/[wId]/builder/assistants/new.tsx [hunk 5]
```diff
@@ -1,27 +1,47 @@
...
+  AssistantBuilderInitialState,
   BUILDER_FLOWS,
   BuilderFlow,
 } from "@app/components/assistant_builder/AssistantBuilder";
 import { getApps } from "@app/lib/api/app";
+import { getAgentConfiguration } from "@app/lib/api/assistant/configuration";
 import { getDataSources } from "@app/lib/api/data_sources";
 import { Authenticator, getSession, getUserFromSession } from "@app/lib/auth";
 
+import { buildInitialState } from "./[aId]";
```
Comment (spolu) [inline front/pages/w/[wId]/builder/assistants/new.tsx:25]: let's make it @app based?

**Review by spolu — APPROVED** [review]
LGTM

Maybe some more code can be dried? If not :+1:

File: front/pages/w/[wId]/builder/assistants/new.tsx (hunk 5 above)
Comment (lasryaric) [reply front/pages/w/[wId]/builder/assistants/new.tsx:25]: For this case I find it cleaner to have a relative import. They are tight together but creating an extra file just for that is a bit too much. I am ok with moving it, but just for the rational.

File: front/pages/w/[wId]/builder/assistants/[aId]/index.tsx (hunk 3 above)
Comment (lasryaric) [reply front/pages/w/[wId]/builder/assistants/[aId]/index.tsx:182]: The data sources are used even when not copying, so it does not fit into `buildInitialState()`.
I think it makes more sense to duplicate it.

**Pull request merged**